plugins/inputs/arista_cloudvision_telemtry/arista_cloudvision_telemetry.go

```diff
@@ -0,0 +1,644 @@
+		}
+	}
+
+	if time.Duration(c.Redial).Nanoseconds() <= 0 {
+		return fmt.Errorf("redial duration must be positive")
+	}
+
+	go func() {
+		for {
+			//Create a map of devices.  This should read device:target.  The target is the same as the serial number to CVP.
+			var cvdevs []string
+
+			for cvpdevice, devicetarget := range c.CvpDevices() {
+				if _, ok := c.devices[devicetarget]; !ok {
+					c.devices[devicetarget] = cvpdevice
+					cvdevs = append(cvdevs, devicetarget)
+				}
+			}
+			if len(cvdevs) != 0 {
+				// Validate configuration
+				var request []*gnmiLib.SubscribeRequest
+				if request, err = c.newSubscribeRequest(cvdevs); err != nil {
+					fmt.Println(err)
+				}
+
+				// Start the request for gNMI interface on CVP.
+				c.wg.Add(len(cvdevs))
+				CvpAddr := c.Cvpaddress
+				for i := range request {
+					// Loop through the requests for targets.
+					thisReq := make([]*gnmiLib.SubscribeRequest, 1)
+					copy(thisReq, (request[i:(i + 1)]))
+					go func(CvpAddr string) {
+						defer c.wg.Done()
+						for ctx.Err() == nil {
+							if err := c.subscribeGNMI(ctx, CvpAddr, tlscfg, thisReq[0]); err != nil && ctx.Err() == nil {
+								acc.AddError(err)
+							}
+
+							select {
+							case <-ctx.Done():
+							case <-time.After(time.Duration(c.Redial)):
+							}
+						}
+					}(CvpAddr)
+				}
+			}
+			time.Sleep(1000 * 1000 * 1000 * 15)
+		}
+	}()
+	return nil
+}
+
+// Method to return all devices which are streaming so we can then use their targets as the gNMI target.
+func (c *CVP) CvpDevices() map[string]string {
+	var bearer = "Bearer " + c.Cvptoken
+	//Connect to CVP resource api
+	req, err := http.NewRequest("GET", "https://"+c.Cvpaddress+"/api/resources/inventory/v1/Device/all", nil)
+	if err != nil {
+		c.Log.Error("Cannot connect to CVP", err)
+	}
+	req.Header.Add("Authorization", bearer)
+	req.Header.Add("Accept", "application/json")
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		c.Log.Error("Cannot connect to CVP", err)
+	}
+	defer resp.Body.Close()
+
+	responseData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		c.Log.Error("Cannot marshall data", err)
+	}
+
+	f := strings.Split(string(responseData), "\n")
+	//Create a map for devices
+	devs := map[string]string{}
+	//Loop through and add devices to devs map that are currently streaming.
+	for _, i := range f {
+		var dev CvPDevices
+		err = json.Unmarshal([]byte(i), &dev)
+		if err != nil {
+			c.Log.Debugf("Cannot marshall HTTP Connection to CVP")
+		}
+		if dev.Result.Value.StreamingStatus == "STREAMING_STATUS_ACTIVE" {
+			devs[dev.Result.Value.Fqdn] = dev.Result.Value.Key.DeviceID
+		}
+	}
+	//Return devices.
+	return devs
+}
+
+//ParsePath from XPath-like string to gNMI path structure
+func parsePath(origin string, pathToParse string, target []string) ([]*gnmiLib.Path, error) {
```

Comment:
I am not sure what you mean by this?  The gnmi input uses this as well.   The only difference is that we are returning a slice of gnmipath for this.  This will generate a slice of paths based on targets and origin.  We can rename to generate paths if that makes more sense.